Userland/DynamicLoader/main.cpp

```diff
@@ -76,6 +80,30 @@ NAKED void _start(int, char**, char**)
 #endif
 }
 
+static ErrorOr<int> open_executable(char const* path)
+{
+    int rc = open(path, O_RDONLY | O_EXEC, 0666);
```

Comment:
The mode argument to open is only relevant if you are using O_CREAT. Using 0666 here is misleading, because it is an unused argument.